internal/services/formatters/generic/trivy/formatter.go

```diff
@@ -125,55 +117,83 @@ func (f *Formatter) setVulnerabilities(cmd Cmd, result *entities.Result, path st
 	}
 }
 
-// nolint:funlen // setVulnerabilitiesOutput is necessary more 15 lines
 func (f *Formatter) setVulnerabilitiesOutput(result []*types.DetectedVulnerability, target string) {
 	for _, vuln := range result {
-		addVuln := &vulnerability.Vulnerability{
-			VulnerabilityID: uuid.New(),
-			Line:            "0",
-			Column:          "0",
-			Confidence:      confidence.Medium,
-			File:            target,
-			Code:            vuln.PkgName,
-			Details:         fmt.Sprintf("%s\n%s\n%s", vuln.Description, vuln.PrimaryURL, getDetails(vuln)),
-			SecurityTool:    tools.Trivy,
-			Language:        languages.Generic,
-			Severity:        severities.GetSeverityByString(vuln.Severity),
-			Type:            enumsVulnerability.Vulnerability,
-		}
+		addVuln := f.getVulnBase()
+		addVuln.File = target
+		addVuln.Code = vuln.PkgName
+		addVuln.Details = f.getDetails(vuln)
+		addVuln.Severity = severities.GetSeverityByString(vuln.Severity)
 		addVuln = vulnhash.Bind(addVuln)
 		f.AddNewVulnerabilityIntoAnalysis(addVuln)
 	}
 }
 
-func getDetails(vuln *types.DetectedVulnerability) string {
-	basePath := "https://cwe.mitre.org/data/definitions/"
-	var result string
+func (f *Formatter) getDetails(vuln *types.DetectedVulnerability) string {
+	details := f.getBaseDetailsWithoutCWEs(vuln)
 
-	for _, id := range vuln.CweIDs {
-		idAfterSplit := strings.SplitAfter(id, "-")
-		result = result + basePath + idAfterSplit[1] + ".html\n"
+	if len(vuln.CweIDs) > 0 {
+		return f.getDetailsWithCWEs(details, vuln)
 	}
-	return result
+
+	return strings.TrimRight(details, "\n")
+}
+
+func (f *Formatter) getBaseDetailsWithoutCWEs(vuln *types.DetectedVulnerability) (details string) {
+	if vuln.Description != "" {
+		details += vuln.Description + "\n"
```

Comment:
I would prefer to use [`strings.Builder`](https://pkg.go.dev/strings#Builder) instead concatenating strings. But I can live with this.

Reply:
I think the idea is important, but the entire project is using this format. I'm going to create an issue with this improvement point so we can leave all formatters standardized.